### PR TITLE
define strong-central docker image layout

### PR DESCRIPTION
**Pull request by rmg**
The image layout is given in the form of a Dockerfile, but the
intention is not to require a Dockerfile to generate the image as long
as it provides the same capabilities and interface.

Connect to strongloop-internal/scrum-nodeops#925

@kraman @sam-github please take a look

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+# Environment variable to be overridden at run/start time
+ENV MESHDB_URL="please set me"
+
+# Define what is run by `docker run strongloop/strong-central`
+# Additional CLI arguments can be appended at run-time
+ENTRYPOINT [\
+    "/usr/local/bin/node", \
+    "/var/lib/strong-central/bin/sl-central.js", \
+    "--base", "/data", \
+    "--listen", "8701" \
```
Comment (kraman) [inline Dockerfile]: Assuming this will change when you add support for HTTPS in central

**Comment by kraman** [comment]
Container spec LGTM

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,66 @@
+FROM node:4
```
Comment (sam-github) [inline Dockerfile]: we'll be able to raise our hands when people ask who is running node v4 in production

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ && chown -R strong-central:strong-central /data /certs
+
+# Run everything from this point forward in the context of the
+# unprivileged strong-central user, inside its HOME directory
+WORKDIR /var/lib/strong-central
+ENV HOME=/var/lib/strong-central
+USER strong-central
+
+# Add the current directory (strong-central source) to the container
+COPY . /var/lib/strong-central/
```
Comment (sam-github) [inline Dockerfile]: Including `node_modules/tap` and `test/`? And non-production deps of the dependencies of central?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+# Install production dependencies
+RUN npm install --production --registry=http://ci.strongloop.com:4873/ \
```
Comment (sam-github) [inline Dockerfile]: I see, the COPY occurs pre-install, so it won't include node_modules, but will include `test/` (I think). Disk is cheap, I guess, but maybe there should be a `.dockerignore` file?

Comment (sam-github) [inline Dockerfile]: This always builds a staging version of the central docker, will we want to have a production version?

File: Dockerfile (hunk 1 above)
Comment (sam-github) [inline Dockerfile]: shouldn't we run with sl-run, or will there be an external restarter? here we are, telling people to deploy with a process manager... and we don't. perhaps we should deploy with strong-pm, so we can troubleshoot problems? Eating our dogfood, and all...

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+    "--listen", "8701" \
+]
+
+## Expected invocation:
+## docker run --detach --publish-all \
+##   --name strong-central-staging \
+##   -e MESHDB_URL=postgres://CREDS@URL/DB \
+##   -e ADDITION_ENVS=values \
+##   -v /staging/data:/data \
+##   -v /staging/certs:/certs:ro \
```
Comment (sam-github) [inline Dockerfile]: certs wll be unencrypted, I guess? Probably no better way, just want to know.

File: Dockerfile (hunk 1 above)
Comment (rmg) [inline Dockerfile]: I thought about supervisor, but then we have the problem of what it connects to... and the problem with PM is, of course, that it has no `run` mode ;-)

File: Dockerfile (hunk 3 above)
Comment (rmg) [inline Dockerfile]: `node_modules` was already being filtered out by the `.dockerignore` file. Added `test/` to it in 869ff70.

File: Dockerfile (hunk 4 above)
Comment (rmg) [inline Dockerfile]: This Dockerfile is mainly for review. My intention is to use [packer](https://packer.io/docs/builders/docker.html) to build the actual images, which doesn't use Dockerfiles. One of the advantages of that is the ability to make some of these config options easier to parameterize.

File: Dockerfile (hunk 5 above)
Comment (rmg) [inline Dockerfile]: If they are encrypted then we will need to provide the password along side them. I'm not sure what the best approach is when getting down to this level. At which point can we trust something enough to access the actual certs?

File: Dockerfile (hunk 1 above)
Comment (sam-github) [inline Dockerfile]: ok

File: Dockerfile (hunk 4 above)
Comment (sam-github) [inline Dockerfile]: ok

File: Dockerfile (hunk 5 above)
Comment (sam-github) [inline Dockerfile]: With h/w encryption the key can be hidden, so its not recoverable, though an attacker can use the h/w to decrypt and sign, in theory at least. I'm OK with this.

**Comment by sam-github** [comment]
LGTM

**Pull request merged**